src/web/user_agent.rs

```diff
@@ -58,12 +58,28 @@ impl DeviceInfo {
         self.os_family == OsFamily::IOs && self.ff_version < 36
             || self.os_family != OsFamily::IOs && self.ff_version < 91
     }
+
+    /// Determine if the device is a mobile phone based on either the form factor or OS.
+    pub fn is_mobile(&self) -> bool {
+        matches!(&self.form_factor, FormFactor::Phone | FormFactor::Tablet)
+            || matches!(&self.os_family, OsFamily::Android | OsFamily::IOs)
+    }
 }
 
 /// Parse a User-Agent header into a simplified `DeviceInfo`
 pub fn get_device_info(ua: &str) -> HandlerResult<DeviceInfo> {
-    let wresult = Parser::new().parse(ua).unwrap_or_default();
-
+    let mut wresult = Parser::new().parse(ua).unwrap_or_default();
+
+    // NOTE: Firefox on iPads report back the "desktop" UA
```

Comment:
It's Safari on Desktop isn't it?

```suggestion
    // NOTE: Firefox on iPads report back the Safari "desktop" UA
```